src/lib/authui/sssd/main_dialog.rb

```diff
@@ -0,0 +1,460 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "auth/authconf.rb"
+require "authui/sssd/uidata.rb"
+require "authui/sssd/new_section_dialog.rb"
+require "authui/sssd/manage_ad_dialog.rb"
+require "authui/sssd/edit_param_dialog.rb"
+require "authui/sssd/extended_param_dialog.rb"
+
+module SSSD
+    # Main dialog is split into overview on left side, and config editor on right side.
+    class MainDialog
+        include Yast
+        include Auth
+        include UIShortcuts
+        include I18n
+        include Logger
+
+        def initialize
+            textdomain "auth-client"
+            # SSSD section name to UI caption mapping
+            @sect_name_caption = {
+                "sssd" => _("Global Options"), "nss" => _("Name switch"),
+                "pam" => _("Authentication"), "sudo" => _("Sudo"),
+                "autofs" => _("Auto-Mount"), "ssh" => _("SSH Public Keys"),
+                "pac" => _("Privilege Account Certificate (MS-PAC)"),
+            }
+            # The reverse caption mapping
+            @sect_caption_name = Hash[*@sect_name_caption.map { |name, caption| [caption, name] }.flatten]
+        end
+
+        def run
+            return if !render_all
+            begin
+                return ui_event_loop
+            ensure
+                UI.CloseDialog()
+            end
+        end
+
+    private
+
+            # Render overview and edit buttons on left side, config editor on right side.
+            def render_all
+                UI.OpenDialog(
+                    Opt(:decorated, :defaultsize),
+                    VBox(
+                        Left(HBox(
+                            Icon::Simple("yast-nis"),
+                            Heading(_("Manage Domain User Logon"))
```

Comment:
Usage of the term logon is consistent with terminology used in Microsoft Windows.